spirv_cross_c.cpp

```diff
@@ -992,6 +1012,34 @@ spvc_result spvc_compiler_msl_add_vertex_attribute(spvc_compiler compiler, const
 #endif
 }
 
+spvc_result spvc_compiler_msl_add_vertex_attribute_2(spvc_compiler compiler, const spvc_msl_vertex_attribute_2 *va)
```

Comment:
I see that the vertex shaders are still using attributes in the compute kernel variant. Is vecsize actually needed? Doesn't attribute deal with it in compute?

Reply:
Like I said above, this is used to fill in the holes in the input buffer in a tessellation control shader, in case an output from the previous stage is not declared. We have to do this in order to make sure the `struct` layout matches.

Reply:
Ok, so the general idea here is to first look at the vertex shader, figure out all its outputs and capture their types as well.

Then, we want to replicate the exact buffer layout for tessellation control inputs?

Reply:
That might be a workable strategy.

Reply:
> What happens if the application is expecting to load a float2, but the type was mapped to float4 in this scheme?

If there is a next member, its offset should be set as if the previous member were a `float4`.

Reply:
Do you want to keep using vecsize in that case, or use offset directly?

Reply:
Actually, it turns out I need to do this for fragment shaders, too. Unlike Vulkan, Metal complains when the types don't match exactly. I'll try to make a separate change.